src/infrastructure/discord/DiscordService.ts

```diff
@@ -18,11 +31,39 @@ export class DiscordService
   private onUserLeaveHandlers: ((
     member: GuildMember | PartialGuildMember,
   ) => void)[] = [];
+  private onVoiceEventHandlers: ((
+    event: DiscordGuildScheduledEvent,
+  ) => void)[] = [];
 
   constructor(client: Client) {
     this.client = client;
   }
 
+  private convertDiscordEvent(
+    event: GuildScheduledEvent | PartialGuildScheduledEvent,
+  ): DiscordGuildScheduledEvent {
+    // Mapear status do Discord para nosso tipo customizado
+    const statusMap: Record<string, DiscordEventStatus> = {
```

Comment:
o tipo inferido como constante é mais estrito do que `Record` porque só aceita 1,2,3,4
```suggestion
    const statusMap = <const>{
```